From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MemeWeft <165016734+MemeWeft@users.noreply.github.com>
Date: Sat, 16 May 2026 12:28:24 +0200
Subject: [PATCH] Memegot-Startup-Boot


diff --git a/net/minecraft/server/dedicated/DedicatedServer.java b/net/minecraft/server/dedicated/DedicatedServer.java
index eb06d8f012684845146429832e977e6c1ddcd62b..4e38e9d131f00feb5ecdaada55d2903447044add 100644
--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -261,6 +261,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         this.paperConfigurations.initializeGlobalConfiguration(this.registryAccess());
         this.paperConfigurations.initializeWorldDefaultsConfiguration(this.registryAccess());
         // Paper end - initialize global and world-defaults configuration
+        me.memeweft.paper.Memegot.boot(); // Memegot - boot
         this.server.spark.enableEarlyIfRequested(); // Paper - spark
         // Paper start - fix converting txt to json file; convert old users earlier after PlayerList creation but before file load/save
         if (this.convertOldUsers()) {
@@ -905,6 +906,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
     @Override
     public void stopServer() {
         this.notificationManager().serverShuttingDown();
+        me.memeweft.paper.Memegot.shutdown(); // Memegot - shutdown
         super.stopServer();
         //Util.shutdownExecutors(); // Paper - Improved watchdog support; moved into super
     }
